### ISSUE TITLE
`migrate upgrade_repo_v2` fails with unclear error on v2 repos

### ISSUE THREAD

**Issue by tbm** [restic/restic#4114]
Output of `restic version`:  restic 0.14.0 compiled with go1.19 on linux/amd64 (from GitHub)

How did you run restic exactly? `restic migrate upgrade_repo_v2`

What backend/server/service did you use to store the repository? Local storage

I'm trying to upgrade a repo to v2 by following the instructions from https://restic.readthedocs.io/en/stable/045_working_with_repos.html#upgrading-the-repository-format-version

First, `restic migrate upgrade_repo_v2` failed with `migration upgrade_repo_v2 cannot be applied: check failed`.  So I ran `restic check` which showed some errors.  I ran `restic prune` and now `restic check` finds no more errors.  However, `migrate upgrade_repo_v2` still says `check failed`, without giving any info what failed and `restic check` works fine.

BTW, this restic backup might be v2 already (I don't know how to check), so my hypothesis is that you forgot to account for that scenario.

If this is not the case, what other info do you need from me?

**Update**: I tested my hypothesis and I think it's correct.

I created a new restic repo and immediately did `migrate upgrade_repo_v2`: `migration upgrade_repo_v2 cannot be applied: check failed`

If the repo is v2 already, you should print an error saying `repo is v2 already`.

Log:

```
65710:tbm@jirafa: ~/d] restic init
created restic repository 77811d39da at /home/tbm/d/test

Please note that knowledge of your password is required to access
the repository. Losing your password means that your data is
irrecoverably lost.
65711:tbm@jirafa: ~/d] restic check
using temporary cache in /tmp/restic-check-cache-2331301686
repository 77811d39 opened (repository version 2) successfully, password is correct
created new cache in /tmp/restic-check-cache-2331301686
create exclusive lock for repository
load indexes
check all packs
check snapshots, trees and blobs
[0:00]          0 snapshots
no errors were found
65712:tbm@jirafa: ~/d] restic migrate upgrade_repo_v2
repository 77811d39 opened (repository version 2) successfully, password is correct
created new cache in /home/tbm/.cache/restic
found 12 old cache directories in /home/tbm/.cache/restic, run `restic cache --cleanup` to remove them
migration upgrade_repo_v2 cannot be applied: check failed
If you want to apply this migration anyway, re-run with option --force
```

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------

restic is so awesome.  I've had a number of data loss issues in the last year or two (at least 2 broken hard drives; some filesystem corruption) and restic helped me recover some of the data (only some because my backups strategy is still not complete, but that's not restic's fault but my own laziness).  I'm very excited about compression support!  Thank you!

**Comment by tbm** [comment]
> BTW, this restic backup might be v2 already (I don't know how to check),

Oh, it says it right here:

> repository 6a887f9c opened (repository version 2) successfully, password is correct

I don't know why I never noticed this.

So, yes, upgrade_repo_v2 gives `check failed` on an v2 repo instead of saying that the repo is v2 already.

**Comment by MichaelEischer** [comment]
Already fixed by #3913.

**Comment by tbm** [comment]
Great!  I looked at existing issues but didn't find anything.  I didn't look at commits.

Maybe it's time for a new point release soon.

Thanks for your work!
